docker/set_build.py

```diff
@@ -10,26 +10,26 @@
 log.setLevel(logging.DEBUG)
 
 
-def get_build_type(travis_tag=None):
-    if not travis_tag:
+def get_build_type(ci_tag=None):
+    if not ci_tag:
         return "qa"
-    log.debug("getting build type for tag: \"%s\"", travis_tag)
-    if re.match(r'v\d+\.\d+\.\d+rc\d+$', travis_tag):
+    log.debug("getting build type for tag: \"%s\"", ci_tag)
+    if re.match(r'v\d+\.\d+\.\d+rc\d+$', ci_tag):
         return 'rc'
-    elif re.match(r'v\d+\.\d+\.\d+$', travis_tag):
+    elif re.match(r'v\d+\.\d+\.\d+$', ci_tag):
         return 'release'
     return 'qa'
 
 
 def main():
     root_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
-    build_type_path = os.path.join(root_dir, 'lbry', 'build_type.py')
+    build_type_path = os.path.join(root_dir, 'lbry', 'lbry', 'build_type.py')
     log.debug("configuring build type file: %s", build_type_path)
-    travis_commit = os.environ['TRAVIS_COMMIT'][:6]
-    build_type = get_build_type(os.environ.get('TRAVIS_TAG', None))
-    log.debug("setting build type=%s, build commit=%s", build_type, travis_commit)
+    commit_hash = os.getenv('CI_COMMIT_SHA', os.getenv('TRAVIS_COMMIT', None))[:6]
+    build_type = get_build_type(os.getenv('CI_COMMIT_TAG', os.getenv('TRAVIS_TAG', None)))
+    log.debug("setting build type=%s, build commit=%s", build_type, commit_hash)
     with open(build_type_path, 'w') as f:
-        f.write(f"BUILD = \"{build_type}\"\nBUILD_COMMIT = \"{travis_commit}\"\n")
+        f.write(f"BUILD = \"{build_type}\"\nBUILD_COMMIT = \"{commit_hash}\"\n")
```

Comment:
You don't need to pass `None` as second argument to `getenv()`  since that's the default already.

For any of the environment variables that are actually required it's better to access them via `os.environ` (throws `KeyError` if variable is missing) instead of `os.getenv()` (always returns None if variable is missing) as this will produce much more useful errors when they are missing.

For example `os.getenv('TRAVIS_COMMIT', None)[:6]` would fail with an error about slicing `None` (not very useful) instead of with an error that the variable is missing (much more useful).

Reply:
good idea. can you use os.getenv and os.environ together? i think i need to make this a bit more verbose to really do what i want

Reply:
My comment is mostly an FYI, I'd like to drop this script altogether; so as long as it works for now that's probably good enough.